cmd/upgrade/cluster/cmd.go

```diff
@@ -428,6 +430,31 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// check if the cluster upgrade requires gate agreements
+	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
```

Comment:
@vkareh @pvasant I wanted to add this call earlier for STS clusters before when we check if a role upgrade is required but wasn't sure we need to prompt the user on manual mode.

Also, that would really mess up the code. In what cases of sts do we prompt the user?

Reply:
@zgalor  why cant we check after the version is selected

Reply:
Because we use the upgrade policy API we need all the fields

Reply:
@zgalor  ok  I was thinking something like this  wdyt? 
```

	upgradePolicyBuilder := cmv1.NewUpgradePolicy().
		ScheduleType("manual").
		Version(version)

	upgradePolicy, err := upgradePolicyBuilder.Build() 
// check if the cluster upgrade requires gate agreements
	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
	if err != nil {
		reporter.Errorf("Failed to check for missing gate agreements upgrade for cluster '%s': %v", clusterKey, err)
		os.Exit(1)
	}
upgradePolicyBuilder.NextRun(nextRun)
```